plugins/chain_plugin/include/eosio/chain_plugin/chain_plugin.hpp

```diff
@@ -293,6 +295,116 @@ class read_only {
       }
    }
 
+   static uint64_t get_table_index_name(const read_only::get_table_rows_params& p) {
+      // see multi_index packing of index name
+      const uint64_t table = p.table;
+      uint64_t index = table & 0xFFFFFFFFFFFFFFF0ULL;
+      EOS_ASSERT( index == table, chain::contract_table_query_exception, "Unsupported table name: ${n}", ("n", p.table) );
+
+      uint64_t pos = 0;
+      if (p.index_position.empty() || p.index_position == "first" || p.index_position == "primary") {
```

Comment:
This is only called when we are explicitly not talking about the primary key.  However, the naming of the method and the operation would imply that there is no difference between the primary and secondary indices.  (They both return 0 for the index byte).  

Can we rename this such that it is obvious that this only works on the non-primary indices and let "first", "primary" etc fall through to the `throw` when it is misused?